### ISSUE TITLE
ChatSecure keep crashing at launch.

### ISSUE THREAD

**Issue by pommm** [ChatSecure/ChatSecure-iOS#827]
Hi,

Our company uses XMPP on Prosody server.  Our employees are having trouble using ChatSecure on iOS. ChatSecure invariably crashes.  Users on Android devices use Conversations with XMPP on Prosody and it runs perfectly well.

User1 is using iPhone 7 plus iOS10.1.1(14B100) ChatSecure 4.1.0. Always crashes at launch app.
User2 is using iPhone 6 iOS10.3.2(14F89) ChatSecure 4.1.0. Always crashes at launch app.
User3 using iPhone 7 plus iOS 10.3.2(14F89) ChatSecure 4.1.0. Rarely crashes like 1 out of every 30 times.

User1 tried upgrading to iOS 10.3.3 (14G60) current version on July 30, 2017, but ChatSecure 4.1.0 still crashes at lauch of the app.

Here is prosody module enabled on our server:

		"roster"; 
		"saslauth"; 
	 	"tls";
		"dialback"; 
		"disco"; 
		"posix"; 
		"mam"; 			
		"mam_archive";
		"private"; 
		"vcard"; 
		"compression";
		"version";
		"uptime";
		"time";
		"ping";
		"pep";
		"register";
		"admin_adhoc";
		"admin_telnet";
		"bosh";
		"http";
		"http_files";
		"http_upload";
		"groups";
		"announce";
		"welcome";
		"watchregistrations";
		"message_logging";
		"smacks";
                "pubsub";
                "cloud_notify";
                "pinger";
                "smacks_offline";
		"carbons";
		"csi";
		"throttle_presence";
		"log_auth";

Here is a screen shot of a ChatSecure crash report. For many of the crashes, we have submitted the crash report. However, we have not received any feedback from the ChatSecure development team or from Apple.   Please help us resolve this problem.  

![crash](https://user-images.githubusercontent.com/26763496/28766394-94b4a248-75f9-11e7-89e1-21e5947c921a.jpg)

Thank you for your kind attention.

**Comment by chrisballinger** [comment]
If you have multiple active group chats you may be experiencing a known crash

**Comment by mrtnhs** [comment]
Same here. Crashes very often at launch on iPhone 6. Sometimes it freezes if I'll try to start a chat. 
A little bit better on my iPhone 7 but not very usable at the moment.
The 4.0.4 version was much better in this behavior.

**Comment by mrtnhs** [comment]
I don't mean groupchats. It's the behavior on a chat with a single person.

**Comment by chrisballinger** [comment]
Can you try the v4.1.1 (98) beta?

**Comment by mrtnhs** [comment]
I gave it s try. It's really better on iPhone 7. 
But not the iPhone 6. Tapping on ones icon to start a chat isn't followed by opening the chatwindow. Only sometimes and only after a long time.

**Comment by chrisballinger** [comment]
Do you have an extremely large chat histroy with that contact? Or other contacts? How about media messages?

**Comment by mrtnhs** [comment]
No, I don't have very large history files because I'm not completely migratet to XMPP.
And only 6 pictures. Tried to delete them but this isn't possible. Chattopics I can delete but not the pictures.

**Comment by chrisballinger** [comment]
I think I've found the crash, or at least one that happens when you receive group chat invitations that can cause crash-on-launch issues. Will be fixed in next build.

**Comment by chrisballinger** [comment]
@mrtnhs I don't see any crash reports coming from an iPhone 6. Could you try to submit the crash report (before it crashes again), or temporarily turn on system-wide crash reporting? https://support.apple.com/en-us/HT202100

**Comment by mrtnhs** [comment]
I turned the system-wide crash reporting on. But this isn't what frustrated me, because I don't scroll through the chat history. 
The freezing is it, what makes it unusable at this time. We are trying to chat between iPhone 6 and 7 and this doesn't work at the moment because of freezing on 6. This doesn't happen with 4.0.4

**Comment by chrisballinger** [comment]
@mrtnhs You mean the UI freezes but never crashes?

**Comment by mrtnhs** [comment]
Thats what happens. 
After reading through the issues I tried to scroll through the chat history on iPhone 7 and it crashed. I never did this before. 
The crash report was send to you. 
After I installed thd 4.1.1 the freezing nearly disappeared on the 7 but on the 6 it continues.

**Comment by mrtnhs** [comment]
I just updated to the 4.1.1 (100) on an iPhone 6 and 7. It´s still a crap on the 6. Starting takes a long time and mostly that´s it. Tapping on any icon and nothing happens, it´s annoying. We can´t use it any more. The best performance to me was version 4.0.9 Any possibility to get it back?

**Comment by chrisballinger** [comment]
@mrtnhs Please turn on system level crash reporting and "Share with App Developers" and then make it crash a few more times on the 6: https://support.apple.com/en-us/HT202100 

You should be able to downgrade to an earlier TestFlight version, but I'm not sure which versions are available. However, if this continues to be a problem and I can't see why it's crashing, you may need to delete and reinstall the app on the 6.

**Comment by mrtnhs** [comment]
Hi, I did as you wished. Even on my iPhone 7 there are crashes. 
I downgraded on the 6 to version 4.1 and now it crashes everytime. Really everytime. No chance to do anything.
Analytics is activated. Should I mail it to you or are you informed by Apple?

**Comment by chrisballinger** [comment]
It crashes even if you delete and reinstall the app? Is there anything weird about your phone configuration or language? Hopefully I will see crash reports via Apple, but if I don't there is a way to send them manually (with a bit of work)

**Comment by mrtnhs** [comment]
I deleted the 4.1.1 and installed the 4.1 and it‘s simply bad. So I reinstalled the 4.1.1. 
The configuration of the iPhone I checked again but couldn‘t find anything. 
Shall I send the crash reports via mail?

Chris Ballinger <notifications@github.com> wrote:
It crashes even if you delete and reinstall the app? Is there anything weird about your phone configuration or language? Hopefully I will see crash reports via Apple, but if I don't there is a way to send them manually (with a bit of work)

**Comment by mimi89999** [comment]
@pommm Is this issue still reproducible?
